### ISSUE TITLE
Warning:  could not be started. Please, check BBB-LTI or BBB-Demo.

### ISSUE THREAD

**Issue by ali-alhaidary** [bigbluebutton/bigbluebutton#15387]
BBB v2.5.3
Starting BigBlueButton
Too few arguments.

'# Warning:  could not be started. Please, check BBB-LTI or BBB-Demo.
'#     Run the command:
'#       sudo journalctl -u 
'#     To better understand the ERROR
And when executing   sudo journalctl -u
journalctl: option requires an argument -- 'u'

**Comment by beckerr-rzht** [comment]
I have the same problem and using this temporary fix:
```diff
--- /usr/bin/bbb-conf.org       2022-07-15 12:36:30.402998113 +0200
+++ /usr/bin/bbb-conf   2022-07-15 12:36:55.497166030 +0200
@@ -420,7 +420,7 @@
         TOMCAT_SERVICE=$TOMCAT_USER
     fi
 
-    systemctl start $TOMCAT_SERVICE || {
+    [ -z "$TOMCAT_SERVICE" ] || systemctl start $TOMCAT_SERVICE || {
         echo
         echo "# Warning: $TOMCAT_SERVICE could not be started. Please, check BBB-LTI or BBB-Demo."
         echo "#     Run the command:"
```

**Comment by ali-alhaidary** [comment]
> I have the same problem and using this temporary fix:
> 
> ```diff
> --- /usr/bin/bbb-conf.org       2022-07-15 12:36:30.402998113 +0200
> +++ /usr/bin/bbb-conf   2022-07-15 12:36:55.497166030 +0200
> @@ -420,7 +420,7 @@
>          TOMCAT_SERVICE=$TOMCAT_USER
>      fi
>  
> -    systemctl start $TOMCAT_SERVICE || {
> +    [ -z "$TOMCAT_SERVICE" ] || systemctl start $TOMCAT_SERVICE || {
>          echo
>          echo "# Warning: $TOMCAT_SERVICE could not be started. Please, check BBB-LTI or BBB-Demo."
>          echo "#     Run the command:"
> ```

Thank you @beckerr-rzht , but this is too technical for me :-)

**Comment by ali-alhaidary** [comment]
I think there is a syntax error:
        echo "# Warning: $TOMCAT_SERVICE could not be started. Please, check BBB-LTI or BBB-Demo." <-- here
        echo "#     Run the command:"
        echo "#       sudo journalctl -u $TOMCAT_SERVICE"  <-- and here
        echo "#     To better understand the ERROR"

or, $TOMCAT_SERVICE is blank...

**Comment by ali-alhaidary** [comment]
or, there is something wrong with:
    if systemctl list-units --full -all | grep -q $TOMCAT_USER.service; then
        TOMCAT_SERVICE=$TOMCAT_USER
    fi

**Comment by beckerr-rzht** [comment]
This issue has been fixed 14 days ago in commit https://github.com/bigbluebutton/bigbluebutton/commit/0b0cf2f6bf54d1f74197c4bdd0b70d77e49f335d and is part of `v2.6.x-release`.

This fix should also be ported to `v2.5.x-release`.

**Comment by ali-alhaidary** [comment]
> This issue has been fixed 14 days ago in commit [0b0cf2f](https://github.com/bigbluebutton/bigbluebutton/commit/0b0cf2f6bf54d1f74197c4bdd0b70d77e49f335d) and is part of `v2.6.x-release`.
> 
> This fix should also be ported to `v2.5.x-release`.

So @beckerr-rzht , what should I do?

**Comment by mvasylenko** [comment]
@ali-alhaidary, you can use bbb-conf from develop or v2.6.x-release:
```
wget https://raw.githubusercontent.com/bigbluebutton/bigbluebutton/develop/bigbluebutton-config/bin/bbb-conf -O /usr/bin/bbb-conf
chmod +x /usr/bin/bbb-conf
bbb-conf --clean
```

**Comment by antobinary** [comment]
@GuiLeme is your fix https://github.com/bigbluebutton/bigbluebutton/pull/15307 handling this case?

**Comment by ali-alhaidary** [comment]
Thank you @mvasylenko after applying I get:
# Potential problems described below
grep: /opt/freeswitch/var/log/freeswitch/freeswitch.log: No such file or directory

**Comment by antobinary** [comment]
@ali-alhaidary -- bbb-conf from branch 'develop' or 'v2.6.x-release' will not work on BBB 2.5 at this point. 

I'd recommend pull again the v2.5.x-release version of `bbb-conf`

```
wget https://github.com/bigbluebutton/bigbluebutton/raw/v2.5.x-release/bigbluebutton-config/bin/bbb-conf -O /usr/bin/bbb-conf
chmod +x /usr/bin/bbb-conf
bbb-conf --clean
```

that should bring you back to stock BBB 2.5.3

And then you could try the suggested patch manually, it just means shortly after line 420 in `/usr/bin/bbb-conf` 
change line
`systemctl start $TOMCAT_SERVICE || {`
to
`[ -z "$TOMCAT_SERVICE" ] || systemctl start $TOMCAT_SERVICE || {`

I have not tried it but it makes sense - only start `tomcat9` if it's an existing service.
____

P.S. (based on the cases I've seen so far) all of this is possibly just noise unless you rely on `bbb-demo`. Is your BBB working?

**Comment by ali-alhaidary** [comment]
Thank you @antobinary , I saved the old file, and restored it. Yes, BBB is working just fine, no problem reported till now.

**Comment by ReimarBauer** [comment]
this seems also to fix my upgrade problem. Thx @antobinary

**Comment by fireba11** [comment]
The workaround didn't help for me. scalelite still thinks the server is offline.
Sadly i upgraded both servers in my test environment at once, so i apparently have to wipe both :-/
At least i cought it in testing.

**Comment by fireba11** [comment]
Update (2.5.2 to 2.5.3) said

Restarting the BigBlueButton 2.5.3 ...
Stopping BigBlueButton

Applying updates in /etc/bigbluebutton/bbb-conf/apply-config.sh:
 - Scale up Kurento
  - Configuring three Kurento Media Servers (listen only, webcam, and screenshare)
Generating a RSA private key
..............................................++++
..++++
writing new private key to '/tmp/dtls-srtp-key.pem'
-----
 - Enable UFW rules
  - Enable Firewall and opening 22/tcp, 80/tcp, 443/tcp and 16384:32768/udp
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Firewall is active and enabled on system startup
 - Setting camera defaults

Starting BigBlueButton
Too few arguments.

# Warning:  could not be started. Please, check BBB-LTI or BBB-Demo.
#     Run the command:
#       sudo journalctl -u
#     To better understand the ERROR

bbb-conf --status shows all services running

I'm still trying to figure out where exactly it broke apart ...

**Comment by GuiLeme** [comment]
@fireba11, this workaround proposed by @antobinary https://github.com/bigbluebutton/bigbluebutton/issues/15387#issuecomment-1185664522, could remove the warning part:
```
# Warning:  could not be started. Please, check BBB-LTI or BBB-Demo.
#     Run the command:
#       sudo journalctl -u
#     To better understand the ERROR
```
And regarding scalelite, could you try and do what you suggested? Update Ubuntu.

And if you still have these problems, please let us know.

**Comment by beckerr-rzht** [comment]
Same problem with Version 2.5.4.

Used the [same patch](https://github.com/bigbluebutton/bigbluebutton/issues/15387#issuecomment-1185419014)  to fix it:
```bash
patch -d / -p0 <<EOF
--- /usr/bin/bbb-conf.org       2022-07-15 12:36:30.402998113 +0200
+++ /usr/bin/bbb-conf   2022-07-15 12:36:55.497166030 +0200
@@ -420,7 +420,7 @@
         TOMCAT_SERVICE=$TOMCAT_USER
     fi  

-    systemctl start $TOMCAT_SERVICE || {
+    [ -z "$TOMCAT_SERVICE" ] || systemctl start $TOMCAT_SERVICE || {
         echo
         echo "# Warning: $TOMCAT_SERVICE could not be started. Please, check BBB-LTI or BBB-Demo."
         echo "#     Run the command:"
EOF
```

**Comment by fireba11** [comment]
strange. with that fix i still get
`bbb-conf --clean
Restarting BigBlueButton 2.5.2 (and cleaning out all log files) ...
Stopping BigBlueButton
 ... cleaning log files

Applying updates in /etc/bigbluebutton/bbb-conf/apply-config.sh:
 - Scale up Kurento
  - Configuring three Kurento Media Servers (listen only, webcam, and screenshare)
Generating a RSA private key
................................................................................................................................................................................................................................................................................................................................................................++++
........................................................................................................................................................................................................................................................................................................++++
writing new private key to '/tmp/dtls-srtp-key.pem'
-----
 - Enable UFW rules
  - Enable Firewall and opening 22/tcp, 80/tcp, 443/tcp and 16384:32768/udp
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Skipping adding existing rule
Skipping adding existing rule (v6)
Firewall is active and enabled on system startup
 - Setting camera defaults

Starting BigBlueButton


# Potential problems described below
# Not running:  tomcat9 or grails
................................................................................
# Error: Could not connect to the configured hostname/IP address
#
#    https://0.bbb-test.domain.com/
#
# If your BigBlueButton server is behind a firewall, see FAQ.`
even with 2.5.2 which worked fine earlier. even spun up the old 2.4.x instance to verify it still works with all updates installed -_-

**Comment by beckerr-rzht** [comment]
Did you try, if `bbb-conf --setip ...` fixes this problem?
https://docs.bigbluebutton.org/admin/bbb-conf.html#--setip-hostname_or_ip

**Comment by fireba11** [comment]
found the issue ... appart from the error in /usr/bin/bbb-conf i also got issues with
the securitySalt being added to bbb-web.properties that did not make sure it was on a seperate line, messing up the last value i had in there m(

**Comment by ali-alhaidary** [comment]
This issue is still in v2.5.4, bbb is working fine :-)

**Comment by JDFDEV** [comment]
Had the issue as well, and currently the fix that worked was:
1. Did the latest dist-upgrade on ubuntu.
2. Run the install.sh again on top of the current installation.
3. Reboot. 

All systems go!

**Comment by JDFDEV** [comment]
Update:
The error " Warning:  could not be started. Please, check BBB-LTI or BBB-Demo.
" reappears on bbb-conf --restart ONLY. On a reboot or with bbb-conf --check everything is OK.

**Comment by ali-alhaidary** [comment]
still in 2.5.5, the work around works...

**Comment by beckerr-rzht** [comment]
Fix for 2.5.5 as patch command:
```bash
patch -d / -p0 <<EOF
--- /usr/bin/bbb-conf.orig  2022-09-04 00:06:37.593199706 +0200
+++ /usr/bin/bbb-conf   2022-09-04 09:56:07.094349690 +0200
@@ -420,7 +420,7 @@
         TOMCAT_SERVICE=$TOMCAT_USER
     fi

-    systemctl start $TOMCAT_SERVICE || {
+    [ -z "$TOMCAT_SERVICE" ] || systemctl start $TOMCAT_SERVICE || {
         echo
         echo "# Warning: $TOMCAT_SERVICE could not be started. Please, check BBB-LTI or BBB-Demo."
         echo "#     Run the command:"
EOF
```

**Comment by antobinary** [comment]
Closed via https://github.com/bigbluebutton/bigbluebutton/pull/15307
